src/WebJobs.Extensions.DocumentDB/Trigger/CosmosDBTriggerAttributeBindingProvider.cs

```diff
@@ -91,27 +92,55 @@ public Task<ITriggerBinding> TryCreateAsync(TriggerBindingProviderContext contex
+            string connectionString = this._monitorConnectionString;
+            if (!TryReadFromSettings(attribute.ConnectionStringSetting, connectionString, out connectionString))
+            {
+                throw new InvalidOperationException(
+                    string.Format(CultureInfo.CurrentCulture,
+                    "The CosmosDBTriggerAttribute.ConnectionStringSetting '{0}' property specified does not exist in the Application Settings.",
+                    attribute.ConnectionStringSetting));
+            }
+
+            if (string.IsNullOrEmpty(connectionString))
             {
                 throw new InvalidOperationException(
                     string.Format(CultureInfo.CurrentCulture,
-                    "The DocumentDBTrigger connection string must be set either via a '{0}' app setting, via the DocumentDBTriggerAttribute.ConnectionString property or via DocumentDBConfiguration.TriggerConnectionString.",
+                    "The CosmosDBTrigger connection string must be set either via a '{0}' app setting, via the CosmosDBTriggerAttribute.ConnectionStringSetting property or via DocumentDBConfiguration.ConnectionString.",
                     DocumentDBConfiguration.AzureWebJobsDocumentDBConnectionStringName));
             }
 
-            return attribute.ConnectionStringSetting ?? _monitorConnectionString;
+            return connectionString;
         }
 
         private string ResolveAttributeLeasesConnectionString(CosmosDBTriggerAttribute attribute, string triggerConnectionString)
         {
             // If the leases connection is not specified, it connects to the monitored service
-            return attribute.LeaseConnectionStringSetting ?? _leasesConnectionString ?? triggerConnectionString;
+            string connectionString = string.IsNullOrEmpty(this._leasesConnectionString) ? triggerConnectionString : this._leasesConnectionString;
+            if (!TryReadFromSettings(attribute.LeaseConnectionStringSetting, connectionString, out connectionString))
+            {
+                throw new InvalidOperationException(
+                    string.Format(CultureInfo.CurrentCulture,
+                    "The CosmosDBTriggerAttribute.LeaseConnectionStringSetting '{0}' property specified does not exist in the Application Settings.",
+                    attribute.ConnectionStringSetting));
+            }
+
+            return connectionString;
         }
 
         private ChangeFeedHostOptions ResolveLeaseOptions(CosmosDBTriggerAttribute attribute)
         {
             return attribute.LeaseOptions ?? _leasesOptions;
         }
+
+        private static bool TryReadFromSettings(string settingsKey, string defaultValue, out string settingsValue)
+        {
+            if (string.IsNullOrEmpty(settingsKey))
+            {
+                settingsValue = defaultValue;
+                return true;
+            }
+            settingsValue = ConfigurationManager.AppSettings[settingsKey];
```

Comment:
Instead of reading from `ConfigurationManager` directly, use `INameResolver.Resolve(settingsKey)`.  The DefaultNameResolver pulls from App Settings, then falls back to Environment variables. It also makes it easier to test as you can pass in your own test `INameResolver` without needing app settings or environment variables set up.